x/staging/subspaces/keeper/keeper.go

```diff
@@ -0,0 +1,176 @@
+package keeper
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/desmos-labs/desmos/x/staging/subspaces/types"
+)
+
+type Keeper struct {
+	storeKey sdk.StoreKey
+	cdc      codec.BinaryMarshaler
+}
+
+// NewKeeper creates new instances of the subspaces keeper
+func NewKeeper(storeKey sdk.StoreKey, cdc codec.BinaryMarshaler) Keeper {
+	return Keeper{
+		storeKey: storeKey,
+		cdc:      cdc,
+	}
+}
+
+// SaveSubspace saves the given subspace inside the current context.
+// It assumes that the subspace has been validated already.
+func (k Keeper) SaveSubspace(ctx sdk.Context, subspace types.Subspace) {
+	store := ctx.KVStore(k.storeKey)
+	key := types.SubspaceStoreKey(subspace.ID)
+	store.Set(key, k.cdc.MustMarshalBinaryBare(&subspace))
+}
```

Comment:
I think we should check for some things here. For example, with this code it appears to me that the user can save the subspace even without checking if the editor is valid. I think we should at least check if the editor matches the existing owner at least. 

Implementing this check here would make sure that it is performed also from other modules that might call this method (eg. when implementing IBC capabilities here)

Reply:
You're right

Reply:
I'm thinking if it's better to check it outside the save function instead, what do you think?

Reply:
@bragaz Checking outside this method would mean we would have to re-implement the checks in every places we use this method, which would cause duplicated code. I think having validity checks directly inside the Save method is a lot better to preserve a more maintainable code and avoid duplication

Reply:
@RiccardoM try to take a look at it now, I've made some changes also in the other keeper methods that were using the save. I opted to save it with the store to not repetitively calling the `GetSubspace()` function which is nested inside both save and the checks functions.
Alternatively I can remove the `checkSubspaceAndOwner` function, and only getting the subspace and then let the `SaveSubspace` function handle the checks on the owner...what do you think?  Doing like this will eventually ending in use the `GetSubspace` Method twice.. I'm not convinced